test/lint/test_runner/src/main.rs

```diff
@@ -87,6 +95,68 @@ fs:: namespace, which has unsafe filesystem functions marked as deleted.
     }
 }
 
+fn lint_includes_build_config() -> LintResult {
+    let config_path = "./src/config/bitcoin-config.h.in";
+    if !Path::new(config_path).is_file() {
+        assert!(Command::new("./autogen.sh")
+            .status()
+            .expect("command error")
+            .success());
+    }
+    let defines_regex = check_output(Command::new("grep").args(["undef ", "--", config_path]))
```

Comment:
How about this to at least filter out the easy commented lines?
```diff
diff --git a/test/lint/test_runner/src/main.rs b/test/lint/test_runner/src/main.rs
index d9950c3fba..fa5ac0a597 100644
--- a/test/lint/test_runner/src/main.rs
+++ b/test/lint/test_runner/src/main.rs
@@ -106,10 +106,16 @@ fn lint_includes_build_config() -> LintResult {
-    let defines_regex = check_output(Command::new("grep").args(["undef ", "--", config_path]))
-        .expect("grep failed")
-        .lines()
-        .map(|line| {
-            line.split("undef ")
-                .nth(1)
-                .expect(&format!("Could not extract name in line: {line}"))
-        })
-        .collect::<Vec<_>>()
-        .join("|");
+    let defines_regex = format!(
+        "^(?!//).*({})",
+        check_output(Command::new("grep").args(["undef ", "--", config_path]))
+            .expect("grep failed")
+            .lines()
+            .map(|line| {
+                line.split("undef ")
+                    .nth(1)
+                    .expect(&format!("Could not extract name in line: {line}"))
+            })
+            .collect::<Vec<_>>()
+            .join("|")
+    );
+
+    println!("{}", defines_regex);
+
@@ -120 +126 @@ fn lint_includes_build_config() -> LintResult {
-                "--extended-regexp",
+                "--perl-regexp",
```

Reply:
I think this is a reasonable enough escape hatch. Given that we only have 1 currently, requiring a `// FOO` rather than a `/* FOO */` seems ok to me.

I don't see how we could do it as a clang-tidy plugin unfortunately. Its preprocessor stuff kinda sucks, and I believe we'd also have to keep a hard-coded list because any actual `bitcoin-config.h` would be guaranteed to leave _some_ things undef'd. Hmm, I suppose the plugin itself could read the `.h.in` for each invocation. That's pretty ugly though :(